waspc/waspls/src/Wasp/LSP/Analysis.hs

```diff
@@ -22,12 +22,21 @@ import Wasp.LSP.Debouncer (debounce)
 import Wasp.LSP.Diagnostic (WaspDiagnostic (AnalyzerDiagnostic, ParseDiagnostic), waspDiagnosticToLspDiagnostic)
 import Wasp.LSP.ExtImport.Diagnostic (updateMissingExtImportDiagnostics)
 import Wasp.LSP.ExtImport.ExportsCache (refreshExportsForAllExtImports)
+import qualified Wasp.LSP.Prisma.Analyze as Prisma
 import Wasp.LSP.ServerMonads (HandlerM, ServerM, handler, modify, sendToReactor)
 import qualified Wasp.LSP.ServerState as State
+import Wasp.LSP.Util (getWaspDirFromWaspFileUri)
 
 -- | Finds diagnostics on a wasp file and sends the diagnostics to the LSP client.
 diagnoseWaspFile :: LSP.Uri -> ServerM ()
 diagnoseWaspFile uri = do
+  -- Immediately update Prisma entities only when file watching is enabled
+  prismaSchemaWatchingEnabled <- isJust <$> handler (asks (^. State.regTokens . State.watchPrismaSchemaToken))
+  when prismaSchemaWatchingEnabled $ do
+    case getWaspDirFromWaspFileUri uri of
+      Nothing -> logM $ "Couldn't get wasp dir from wasp file uri " ++ show uri
+      Just waspDir -> Prisma.analyzeAndSetPrismaSchema waspDir
```

Comment:
> So far, it would analyze Wasp file at given uri, then it would, if source watching is enabled (what does that mean though? Why would it be disabled?)

It depends on the LSP client if dynamic capabilities are available. They include watching files for changes like our `schema.prisma` file.

> I am curious how what you did fits into this, maybe you can explain it to me a little bit.

I just looked at what Craig did for TS files: 
1. Register watcher for TS/JS files
2. On change -> update server state with ext diagnostics 
3. Publish new diagnostics
Also, update ext file diagnostics when the Wasp file diagnostics are running.

So I did things analogously: watch and update on change + update when Wasp diagnostics are run.